tests/tpch/generate_data.py

```diff
@@ -60,20 +66,75 @@ def generate(
     if use_coiled:
         with coiled.Cluster(
             n_workers=10,
-            # workload is best with 1vCPU and ~3-4GiB memory
-            worker_vm_types=["m7a.medium", "m3.medium"],
+            # workload is best with 1vCPU and ~6-8GiB memory
+            worker_cpu=[1, 2],
+            worker_memory=["6GiB", "8GiB"],
             worker_options={"nthreads": 1},
             region=REGION,
         ) as cluster:
             cluster.adapt(minimum=1, maximum=350)
             with cluster.get_client() as client:
-                jobs = client.map(_tpch_data_gen, range(0, scale), **kwargs)
-                client.gather(jobs)
+                tpch_from_client(client, **kwargs)
     else:
-        _tpch_data_gen(step=None, **kwargs)
+        with dask.distributed.Client() as client:
+            tpch_from_client(client, **kwargs)
+
+            # TODO: hack to suppress noisy "CommClosedError"s
+            # xref: https://github.com/dask/distributed/issues/7192
+            client.retire_workers()
+            time.sleep(5)
     return str(path)
 
 
+def tpch_from_client(client, scale: int, compression: CompressionCodec, **kwargs):
+    jobs = client.map(
+        _tpch_data_gen, range(0, scale), scale=scale, compression=compression, **kwargs
+    )
+    small_tables = client.gather(jobs)
+    client.submit(
+        collect_small_tables, tables=small_tables, compression=compression
+    ).result()
+
+
+def collect_small_tables(
+    tables: list[dict[str, SmallTable]], compression: CompressionCodec
+):
+    table_names = {t for d in tables for t in d}
+    for table in table_names:
+        small_tables: list[SmallTable] = [t[table] for t in tables if table in t]
+        if small_tables:
+            name = small_tables[0].name
+            out_dir = small_tables[0].out_dir
+            table = pa.concat_tables(t.data for t in small_tables)
```

Comment:
Does this approach guarantee the correct ordering? I'm not sure it's relevant here but I'd like the data to look the same as if generated by a single process.

Reply:
Before it 'worked' but I think it's mostly chance to that the futures returned in order and less likely at larger scales. Updated w/ https://github.com/coiled/benchmarks/pull/1392/commits/40e155563d9d01e587b35ffcc02ab5a7d1f7b6a8 to ensure they're returned in order.

Reply:
Thanks!